### ISSUE TITLE
GeoJsonLayer is not rendered properly when using MapboxLayer integration

### ISSUE THREAD

**Issue by asamoshkin-casafari** [visgl/deck.gl#6284]
#### Description
I'm using:
- React, + StaticMap from "react-map-gl"
- Vendor integration: MapboxLayer from "@deck.gl/mapbox"

I have a React app. I'm using Mapbox as a base map, and DeckGL to render several layers on top of base map:
1. GeoJsonLayer (just a rectangular polygon area on the map)
2. ScatterplotLayer (to show an avarage price of real estate properties in particular streets of the given region)
3. IconLayer (just to set few pins on the map)

Layers are supposed to be rendered in the above order.

Since I can render quite a couple of maps on the single page, I'm also using "MapboxLayer" integration "@deck.gl/mapbox" in order to share the same WebGL context between DeckGL and Mapbox. It allows us to render up to 16 maps on the single page (given the browser's restriction of max 16 WebGL contexts), as opposite to max of 8 maps when each of DeckGL and Mapbox using it's own <canvas> and WebGL context.

**The issue is GeoJsonLayer rendering is broken.** It overlaps other layers, despite being instructed to be rendered at the very bottom. It looks like it does not allow any other layers to be rendered on top of it.

Here is a picture:
<img width="698" alt="2021-10-08_11-58-15" src="https://user-images.githubusercontent.com/87312757/136547447-06f5e104-b117-47b1-bf4f-78612cf1d939.png">

Notice that ScatterplotLayer and IconLayer are rendered properly with regards to each other.

#### Expected Behavior
Without [MapboxLayer integration](https://deck.gl/docs/api-reference/mapbox/overview), when each DeckGL and Mapbox using its own <canvas> it works properly (the default approach without any integrations):

<img width="691" alt="2021-10-08_11-58-57" src="https://user-images.githubusercontent.com/87312757/136546625-5c80cbe9-d329-4461-9b05-0b401ad60860.png">

#### Repro Steps
Here's  a playground:
https://codesandbox.io/s/epic-jepsen-n8s3j

You could switch between two solutions in `index.js` to see the difference:

```
import App from "./deckgl_mapbox_same_gl_context";
// import App from "./deckgl_mapbox";
```

#### Environment
 - Framework Version: deck.gl ^8.4.13
 - Browser Version:  94.0.4606.71
 - OS: Mac OS X 10.15.7
 
Dependencies from `package.json`:

```
"react-map-gl": "^5.2.7",
"@deck.gl/core": "^8.4.13",
"@deck.gl/layers": "^8.4.13",
"@deck.gl/mapbox": "^8.4.13",
"@deck.gl/react": "^8.4.13",
```

#### Logs

The only warning is, but it seems irrelevant for this issue:

```
This page appears to be missing CSS declarations for Mapbox GL JS, which may cause the map to display incorrectly. Please ensure your page includes mapbox-gl.css, as described in https://www.mapbox.com/mapbox-gl-js/api/.
```

**Comment by roniartasitinjak** [comment]
I find the same issue, I also try in web kepler.gl, I add some dataset, try to display geojson and point then I edit the style of map, up the label, road, and border, and the result display the map as I want. I didnt know how keplergl's developers do it.

**Comment by Pessimistress** [comment]
This might be caused by drawing each layer in a separate pass when Mapbox calls `MapboxLayer.render`. Each pass recalculates layer index from 0 which negates the effect of `getPolygonOffset`.

**Comment by roniartasitinjak** [comment]
@Pessimistress its solve work for me thanks for the reference of link. **Try setting depthTest as false in parameters prop.**

**Comment by Pessimistress** [comment]
Fixed in v8.6.6
